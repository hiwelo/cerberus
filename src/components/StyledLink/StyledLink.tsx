import React, { FunctionComponent } from 'react';
import { LinkProps } from '../Link/types';
import { LinkElement } from './components';

export const StyledLink: FunctionComponent<LinkProps> = ({
  children,
  ...rest
}) => <LinkElement {...rest}>{children}</LinkElement>;
