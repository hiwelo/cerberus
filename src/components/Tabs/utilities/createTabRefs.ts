import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
