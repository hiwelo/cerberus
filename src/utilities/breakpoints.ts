/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
