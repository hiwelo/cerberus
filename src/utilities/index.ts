export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
