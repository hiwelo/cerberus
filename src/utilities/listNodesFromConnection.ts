/** Describes the generic structure of a GraphQL edge */
type GenericEdge<T> = {
  /** Item from a GraphQL connection object */
  node: T;
  /** Potential typing for the edge added by GraphQL */
  __typename: string;
};

/** Describes the generic structure of a GraphQL connection object */
type GenericConnection<T> = {
  edges: GenericEdge<T>[];
};

/**
 * Returns an array of all nodes contained in the provided GraphQL connection
 * object
 *
 * @param connection GraphQL connection object containing the nodes to return
 */
export function listNodesFromConnection<T>(
  connection?: GenericConnection<T>,
): T[] | undefined {
  return connection?.edges.map(({ node }) => node);
}
