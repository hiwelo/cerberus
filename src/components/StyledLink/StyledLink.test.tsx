import React from 'react';
import { testA11y, testSnapshot } from 'utilities/test';
import { StyledLink } from './StyledLink';

const TestContainer = <StyledLink href="#">Test</StyledLink>;

describe('StyledLink', () => {
  testSnapshot(TestContainer);
  testSnapshot(
    <StyledLink href="#" target="_blank">
      Test with target blank
    </StyledLink>,
    'with target blank',
  );
  testA11y(TestContainer);
});
