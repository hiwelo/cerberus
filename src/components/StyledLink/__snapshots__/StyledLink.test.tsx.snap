// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink should match snapshot 1`] = `
.c0 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1::before {
  position: absolute;
  bottom: 0.1em;
  left: 0;
  right: 0;
  content: '';
  height: 0.15em;
  background-color: #e8c000;
  border-radius: 0.4rem;
}

.c1:focus {
  outline: none;
}

.c1:focus::before,
.c1:hover::before {
  background-color: #8ab0ab;
  bottom: 0;
}

@media (prefers-reduced-motion:no-preference) {
  .c1::before {
    -webkit-transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
    transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
  }
}

<a
  class="c0 c1"
  href="#"
>
  Test
</a>
`;

exports[`StyledLink should match snapshot: with target blank 1`] = `
.c0 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1::before {
  position: absolute;
  bottom: 0.1em;
  left: 0;
  right: 0;
  content: '';
  height: 0.15em;
  background-color: #e8c000;
  border-radius: 0.4rem;
}

.c1:focus {
  outline: none;
}

.c1:focus::before,
.c1:hover::before {
  background-color: #8ab0ab;
  bottom: 0;
}

@media (prefers-reduced-motion:no-preference) {
  .c1::before {
    -webkit-transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
    transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
  }
}

<a
  class="c0 c1"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  Test with target blank
</a>
`;
